### PR TITLE
more simpler changes

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

File: shared/chat/conversation/messages/message-popup/reactionitem.tsx [hunk 1]
```diff
@@ -23,8 +23,8 @@ const ReactionItem = (props: Props) => {
   const topReacjis = _topReacjis.slice(0, 5)
   return (
     <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.container}>
-      {topReacjis.map(r => (
-        <Kb.ClickableBox key="picker" onClick={() => onReact(r.name)} style={styles.clickableBox}>
+      {topReacjis.map((r, idx) => (
+        <Kb.ClickableBox key={r.name ?? idx} onClick={() => onReact(r.name)} style={styles.clickableBox}>
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/message-popup/reactionitem.tsx:27]: these aren't always unique, so we need a fallback

File: shared/chat/conversation/messages/shared-styles.tsx [hunk 2]
```diff
@@ -1,46 +1,45 @@
 import * as Styles from '../../../styles'
 
-const editing = {
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/shared-styles.tsx:3]: its never safe to pull colors at the module level it has to be wrapped in a function that gets called else your light/dark will be incorrect

File: shared/common-adapters/markdown/service-decoration.tsx [hunk 3]
```diff
@@ -174,7 +174,10 @@ const ServiceDecoration = (props: Props) => {
     )
   } else if (parsed.typ === RPCChatTypes.UITextDecorationTyp.link) {
     const link = parsed.link.url
-    const openUrl = link.startsWith('http://') || link.startsWith('https://') ? link : 'http://' + link
+    const openUrl =
+      link.toLowerCase().startsWith('http://') || link.toLowerCase().startsWith('https://')
```
Comment (chrisnojima) [inline shared/common-adapters/markdown/service-decoration.tsx:178]: on master links that are `HTTPS://keybase.io` will generate `http://HTTPS`

File: shared/fs/common/hooks.tsx [hunk 4]
```diff
@@ -42,7 +42,9 @@ const useFsNonPathSubscriptionEffect = (topic: RPCTypes.SubscriptionTopic) => {
   React.useEffect(() => {
     const subscriptionID = Constants.makeUUID()
     dispatch(FsGen.createSubscribeNonPath({subscriptionID, topic}))
-    return () => dispatch(FsGen.createUnsubscribe({subscriptionID}))
```
Comment (chrisnojima) [inline shared/fs/common/hooks.tsx:45]: newer lint wants us to return void and not what dispatch returns, which is what we want anyways

File: shared/fs/common/open-in-system-file-manager.tsx [hunk 5]
```diff
@@ -5,7 +5,7 @@ import * as Types from '../../constants/types/fs'
 import * as FsGen from '../../actions/fs-gen'
 import {fileUIName} from '../../constants/platform'
 import * as Container from '../../util/container'
-import * as Kbfs from '../common'
```
Comment (chrisnojima) [inline shared/fs/common/open-in-system-file-manager.tsx:8]: there's some circular deps in the fs code which this fixes

File: shared/teams/main/index.tsx [hunk 6]
```diff
@@ -190,13 +189,7 @@ type Row = {key: React.Key} & (
   | {team: Types.TeamMeta; type: 'team'}
 )
 
-type State = {
-  sawChatBanner: boolean
```
Comment (chrisnojima) [inline shared/teams/main/index.tsx:194]: not used at all

File: shared/common-adapters/proof-broken-banner.tsx [hunk 7]
```diff
@@ -20,9 +20,13 @@ type ProofBrokenBannerNonEmptyProps = {
...
-    isMobile
-      ? (username: string) => dispatch(ProfileGen.createShowUserProfile({username}))
-      : (username: string) => dispatch(Tracker2Gen.createShowUser({asTracker: true, username})),
+    (username: string) => {
+      if (isMobile) {
+        dispatch(ProfileGen.createShowUserProfile({username}))
+      } else {
+        dispatch(Tracker2Gen.createShowUser({asTracker: true, username}))
+      }
+    },
```
Comment (songgao-zoom) [inline shared/common-adapters/proof-broken-banner.tsx:29]: Interesting -- isn't the original marginally faster?

File: shared/fs/common/open-hoc.tsx [hunk 8]
```diff
@@ -46,6 +46,7 @@ function OpenHOC(Component: React.ComponentType<Props & InjectedProps>): React.C
 
     return <Component {...props} onOpen={destinationPickerGoTo} />
   }
+  return Inner
```
Comment (songgao-zoom) [inline shared/fs/common/open-hoc.tsx:49]: Does this make a difference?

**Review by songgao-zoom — APPROVED** [review]

File: shared/common-adapters/proof-broken-banner.tsx (hunk 7 above)
Comment (chrisnojima) [reply shared/common-adapters/proof-broken-banner.tsx:29]: this type of optimization makes absolutely no difference. this way the typing is easier for ts

File: shared/fs/common/open-hoc.tsx (hunk 8 above)
Comment (chrisnojima) [reply shared/fs/common/open-hoc.tsx:49]: helps with react devtools, else you get like `_c()` as the name

**Pull request merged**